managers/signals.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import Any
+from typing import Set
+
+from django.db.models.signals import m2m_changed
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from employees.models import TaskActivityType
+from managers.models import Project
+from users.models import CustomUser
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(m2m_changed, sender=Project.managers.through)
+def update_user_type(sender: Project, action: str, pk_set: Set, **kwargs: Any) -> None:
+    assert sender == Project.managers.through
+    project = kwargs["instance"]
+    logger.debug(f"Updates on project with id: {sender.pk} for users with id: {pk_set}")
+    if action in ["pre_remove", "post_remove"]:
+        change_user_type_to_employee(pk_set)
+    elif action in ["pre_add", "post_add"]:
+        change_user_type_to_manager(project)
+    else:
+        return
+
+
+def change_user_type_to_manager(project: Project) -> None:
+    for manager in project.managers.all():
+        logger.debug(f"User with user type: {manager.user_type} with id: {manager.pk} in managers")
+        if manager.user_type == CustomUser.UserType.EMPLOYEE.name:
+            logger.debug(
+                f"User with user type: {manager.user_type} and id: {manager.pk} is in managers for project with id: {project.pk}"
+            )
+            manager.user_type = CustomUser.UserType.MANAGER.name
+            manager.full_clean()
+            manager.save()
+            logger.debug(f"User with id: {manager.pk} have changed user type to manager")
+        else:
```

Comment:
this `else` seems unnecessary

Reply:
I created it because once @kbeker  suggest it to me:
https://github.com/Code-Poets/sheetstorm/pull/98#discussion_r270030292

Reply:
The whole function can be limited to single query 
```
project.managers.filter(
    user_type=CustomUser.UserType.EMPLOYEE.name
).update(
   user_type=CustomUser.UserType.MANAGER.name
)
```